### ISSUE TITLE
Missing Rescaling with training-set size?

### ISSUE THREAD

**Issue by ArneNx** [team-approx-bayes/fromp#2]
Comparing your code with the paper, I can't seem to find the rescaling of the loss before functional regularization by the training set size `N` (see e.g. equation 15). 
Is that on purpose or did I just miss it?

**Comment by siddharthswaroop** [comment]
Hi, thanks for your question. Compared to the paper, the loss is scaled by `1/N`. 

For the two terms in Eq. 8 in the paper: 
1. First term: PyTorch automatically does the average (or, per-example) gradient for the new task's loss `\bar{\ell}_t`. 
2. Second term: we define `tau` in the code as scaled by `N` compared to the `\tau` in the paper. So `tau=10` in the code is the same as `\tau=10N` in the paper. This all works because of the overall `1/N` scaling in the code.

**Comment by ArneNx** [comment]
Ok. So value for `tau` (e.g. 10) is actually implicitly `tau/N`?

**Comment by ArneNx** [comment]
The per-example gradient is done for a batch though. But the functional regulariser is on the scale of the whole dataset. Is that taken into account?

**Comment by siddharthswaroop** [comment]
Yes, the value for `tau` is actually implicitly `tau/N`. 

As we consider the entire loss to be per datapoint (ie scaled by `1/N` compared to the paper), the minibatching/per-example gradients all work out. If we had not scaled the loss by `1/N` already, then we would have had to have additional correction terms.

**Comment by ArneNx** [comment]
That makes sense. Thanks for explaining this.
